packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -257,6 +259,68 @@ export default function createInstantSearchManager({
     }
   }
 
+  function hydrateSearchClient(client, results) {
+    if (!results) {
+      return;
+    }
+
+    if (!client._useCache) {
+      // This condition avoid hydrating a `searchClient` different from the
+      // Algolia one. Unless the implementation uses the exact same approach
+      // for the cache (the property `_useCache`). The implementation is brittle
+      // but we don't have a proper way to detect the Algolia client at the moment.
+      return;
+    }
+
+    if (Array.isArray(results)) {
+      hydrateSearchClientWithMultiIndexRequest(client, results);
+      return;
+    }
+
+    hydrateSearchClientWithSingleIndexRequest(client, results);
+  }
+
+  function hydrateSearchClientWithMultiIndexRequest(client, results) {
+    // https://github.com/algolia/algoliasearch-client-javascript/blob/c27e89ff92b2a854ae6f40dc524bffe0f0cbc169/src/AlgoliaSearchCore.js#L232-L240
```

Comment:
Are you referring to the way we need to differentiate the POST requests with this comment? It could be clearer with prior explanations.

Reply:
Nope, it's referring to the computation of the cache key inside the client.

Reply:
Right, an explanation would help to understand!

Reply:
Right, an explanation would help to understand!

Reply:
https://github.com/algolia/react-instantsearch/pull/2537/commits/a96de76af0d50a184308bc661e16303046d2bed7